### PR TITLE
xtensa-build-zephyr.py: install and checksum .elf, .lst and other files

### PR DESCRIPTION
**Pull request by marc-hb**
4 commits = 3 refactorings with no effect + 1 last commit adding the feature. Separate reviews recommended. See commit messages.

Sample new output below. The sha256 is computed on the _uncompressed_
files and _not_ affected by gzip metadata.

```
build-sof-staging
├── sof
│   ├── community
│   │   ├── sof-imx8.ri         sha256=50c5423d2355ef3ed91...
│   │   └── sof-tgl.ri          sha256=6b1d26c12a63de5dfc8...
│   ├── sof-imx8.ldc    sha256=520d365d188d5ef4e907d3dd8c9...
│   └── sof-tgl.ldc     sha256=cdce6ad340d4dc047e71e738eb0...
├── sof-info
│   ├── imx8
│   │   ├── config.gz
│   │   ├── zephyr.elf.gz
│   │   ├── zephyr.lst.gz       sha256=8c83c3fc92df0a871dd...
│   │   ├── zephyr.map.gz
│   │   └── zephyr.strip.gz     sha256=d3ce3d3450c67bb3580...
│   └── tgl
│       ├── boot.mod.gz         sha256=d9c9e82e75fa6d061bf...
│       ├── config.gz
│       ├── main.mod.gz
│       ├── stripped-main.mod.gz        sha256=c367dccca6d...
│       ├── zephyr.elf.gz
│       ├── zephyr.lst.gz       sha256=5474bc5e58a5d000109...
│       ├── zephyr.map.gz
│       └── zephyr.strip.gz     sha256=6285f41c0682b33b7e0...
└── tools
    ├── cavstool.py
    ├── cavstool_client.py
    ├── cavstwist.sh
    ├── mtrace-reader.py
    ├── remote-fw-service.py
    └── sof-logger
```

cc:
- https://github.com/zephyrproject-rtos/zephyr/pull/51736
- https://github.com/zephyrproject-rtos/zephyr/pull/51737
- https://github.com/zephyrproject-rtos/zephyr/pull/51954

### CODE REVIEW

**Comment by marc-hb** [comment]
Looking at the internal QB server, https://sof-ci.01.org/sof-pr-viewer/#/build/PR6501/build10687663 failed only because of the incomplete 01.org migration, the rest looks OK

Everything else is green including https://sof-ci.01.org/sofpr/PR6501/build2124/build and https://sof-ci.01.org/sofpr/PR6501/build2124/devicetest

**Review by lgirdwood — APPROVED** [review]
@aborisovich pls review

**Comment by marc-hb** [comment]
Downgrading to draft because of the filename change requested in https://github.com/zephyrproject-rtos/zephyr/pull/51736. Only a file name change (so far!) requested in https://github.com/zephyrproject-rtos/zephyr/pull/51736, so this SOF PR #6501 is still ready for review despite the draft status.

**Comment by aiChaoSONG** [comment]
everyone can use binutils to strip the elf, maybe we could exclude zephyr.strip.gz to save some disk space?

**Comment by marc-hb** [comment]
@aiChaoSONG I should have explained the purpose more: you cannot `strip` the .elf file thrown away by automated builds. However you can (most of the time) check the build logs, compare the checksum with the one in your own build logs and immediately see that something is off.

**Review by mengdonglin — APPROVED** [review]
@marc-hb Can this checksum feature be exported later as `EraseVariables`?  So sof-bin scripts may use this feature to show these checksum for debugging purpose if needed.

**Comment by marc-hb** [comment]
zephyr.strip checksum can be seen in https://sof-ci.01.org/sofpr/PR6501/build2540/build/mtl_zph_ipc4.txt

File: app/overlays/strip_elf.conf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# The additional stripped .elf files are deterministic.
+# This adds very few files in the build directory and has
+# absolutely no other effect whatsoever.
+# See on-going discussion in
+# https://github.com/zephyrproject-rtos/zephyr/pull/51954
+# and usage in xtensa-build-zephyr.py
+CONFIG_BUILD_OUTPUT_STRIPPED=y
```
Comment (abonislawski) [inline app/overlays/strip_elf.conf:7]: btw is there any reason why this is done in separate overlay?
If you want to simply turn it on by default for SOF+Zephyr please check prj.conf for such cases:
https://github.com/thesofproject/sof/blob/main/app/prj.conf
so there will be no need to modify build script for single config

**Review by aborisovich — APPROVED** [review]
I won't be blocking this because this SHA is good idea for having reproductive builds but I'm not that happy that this feature is placed into the python script xtensa-build-zephyr.py.
We build more and more dependencies on it while it was supposed to be temporary...
I would try to move this SHA calculation to CMake using https://cmake.org/cmake/help/latest/command/file.html#hash .
If we would every want to move to https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html then

File: app/overlays/strip_elf.conf (hunk 1 above)
Comment (marc-hb) [reply app/overlays/strip_elf.conf]: > btw is there any reason why this is done in separate overlay?
If you want to simply turn it on by default for SOF+Zephyr please check prj.conf for such cases:

No reason, great suggestion. Done, thanks!

**Comment by marc-hb** [comment]
>  because this SHA is good idea for having reproductive builds I'm not that happy that this feature is placed into the python script xtensa-build-zephyr.py.

TODO and link to on-going Zephyr discussion added.

**Review by aiChaoSONG — APPROVED** [review]

**Review by abonislawski — APPROVED** [review]

**Pull request merged**